src/parsers/manifest/dash/index.ts

```diff
@@ -28,7 +38,32 @@ export default function parseFromDocument(
 ) : IParsedManifest {
   const root = document.documentElement;
   if (!root || root.nodeName !== "MPD") {
-    throw new Error("document root should be MPD");
+    throw new Error("DASH Parser: document root should be MPD");
   }
   return parseMPD(root, uri);
 }
+
+/**
+ * Parse Xlinks response as an array of Periods.
+ * @param {string} xlinkData
+ * @param {Object} manifestInfos
+ * @returns {Array.<Object>}
+ */
+export function parseDownloadedXLinks(
+  xlinkData : string,
+  manifestInfos : IManifestInfos
+) : IParsedPeriod[] {
+  const wrappedData = "<root>" + xlinkData + "</root>";
+  const dataAsXML = new DOMParser().parseFromString(wrappedData, "text/xml");
+  const periods = dataAsXML.children;
+  const periodsIR : IPeriodIntermediateRepresentation[] = [];
+  for (let i = 0; i < periods.length; i++) {
+    if (periods[i].nodeType === Node.ELEMENT_NODE) {
+      const period = periods[i] as Element;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unnecessary cast. ([typescript:S4325](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4325))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=401&issues=AWd5lz8RVQEMgdy3ZsrU&open=AWd5lz8RVQEMgdy3ZsrU)

[ISSUE_KEY]: AWd5lz8RVQEMgdy3ZsrU